src/server/game/Handlers/SpellHandler.cpp

```diff
@@ -646,6 +646,11 @@ void WorldSession::HandleSpellClick(WorldPacket& recvData)
     if (!unit)
         return;
 
+    if (_player->IsHostileTo(unit))
```

Comment:
Is there any gob/npc hostile to player with spell click? Just to be sure we are not breaking them.
Anyway, nice catch. 😁